ast/cinaps/name.ml

```diff
@@ -31,8 +19,8 @@ let fix_builtin name =
   then name ^ "_"
   else name
 
-let make prefix tys =
+let make prefix targs =
   fix_builtin
     (String.concat ~sep:"_"
        (List.map ~f:Ml.id
-          (prefix @ add_tuple_suffix tys [])))
+          (prefix @ List.fold_right targs ~init:[] ~f:add_targ_suffix)))
```

Comment:
At this point, it could just be `List.map targs ~f:name_of_targ`, and we could leave out the `fold_right`.